### PR TITLE
Functional test for RHBZ1473203 and some adjustments

### PR DESCRIPTION
**Pull request by ldoktor**
This PR adjusts a bit the "get_console" handling and adds functional test for RHBZ1473203.

### CODE REVIEW

File: tests/avocado/avocado_qemu/test.py [hunk 1]
```diff
@@ -123,7 +123,8 @@ def _handle_prompts(session, username, password, prompt, timeout=10,
     password_prompt_count = 0
     login_prompt_count = 0
     last_chance = False
-
+    # Send enter to refresh output (in case session was attached after boot)
+    session.sendline()
```
Comment (apahim) [inline tests/avocado/avocado_qemu/test.py:141]: ok

File: tests/avocado/avocado_qemu/test.py [hunk 2]
```diff
@@ -131,9 +131,9 @@ def _handle_prompts(session, username, password, prompt, timeout=10,
             match, text = session.read_until_last_line_matches(
                 [r"[Aa]re you sure", r"[Pp]assword:\s*",
                  # Prompt of rescue mode for Red Hat.
-                 r"\(or (press|type) Control-D to continue\):\s*$",
-                 r"[Gg]ive.*[Ll]ogin:\s*$",  # Prompt of rescue mode for SUSE.
-                 r"(?<![Ll]ast )[Ll]ogin:\s*$",  # Don't match "Last Login:"
+                 r"\(or (press|type) Control-D to continue\):\s*",
+                 r"[Gg]ive.*[Ll]ogin:\s*",  # Prompt of rescue mode for SUSE.
+                 r"(?<![Ll]ast )[Ll]ogin:\s*",  # Don't match "Last Login:"
```
Comment (apahim) [inline tests/avocado/avocado_qemu/test.py:150]: Sure. And this can always be improved if we face new corner cases.

File: tests/avocado/rhbz1473203_numa_hotplug.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+class TestNumaHotplug(test.QemuTest):
+    """
+    Verifies that "info numa" and "/sys/devices/system/node/" contains
+    correct values before/after inserting memory devices into default
+    and then into 13th numa node.
+
+    Associated bug trackers: RHBZ1473203
+        https://bugzilla.redhat.com/show_bug.cgi?id=1473203
+
+    Fixed in commit kernel commit dc421b200f91930c9c6a9586810ff8c232cf10fc.
```
Comment (apahim) [inline tests/avocado/rhbz1473203_numa_hotplug.py]: commit kernel commit

File: tests/avocado/rhbz1473203_numa_hotplug.py [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+        cmd = "device_add pc-dimm,id=dimm8,memdev=mem8,node=13"
+        res = self.vm.qmp("human-monitor-command", command_line=cmd)
+        self.assertEqual(res["return"], "")
+        exp = [1280] + [256] * 12 + [1280] + [256] * 2
+        self.check_mem_monitor(self.vm.qmp, exp)
+        self.check_mem_console(console, exp)
+        console.close()
+
+    def tearDown(self):
+        self.vm.shutdown()
```
Comment (apahim) [inline tests/avocado/rhbz1473203_numa_hotplug.py]: This test requires a vm image. I'd include both the parameters.yaml file and the instructions to use it.

**Review by apahim — CHANGES_REQUESTED** [review]
@ldoktor just a minor fix in the docstring. Also, I'd include the parameters.yaml since this test requires a VM image. (or some other way to indicate that).

**Comment by ldoktor** [comment]
OK, updated. There is yet another batch of fixes to `get_console`, change to default params, reworked way to add image to qemu command and not insignificant changes to the functional test make it less-error-prone.

File: tests/avocado/test_nec-usb-xhci.py [hunk 5]
```diff
@@ -18,6 +18,7 @@ class TestNecUsbXhci(test.QemuTest):
 
     def setUp(self):
         usbdevice = os.path.join(self.workdir, 'usb.img')
+        self.require_image()
```
Comment (apahim) [inline tests/avocado/test_nec-usb-xhci.py:21]: `self.request_image()`

Comment (ldoktor) [reply tests/avocado/test_nec-usb-xhci.py:21]: sure, sorry I added it on a last minute... I'll fix it in v2

Comment (apahim) [reply tests/avocado/test_nec-usb-xhci.py:21]: no need for v2. I'm handling that while integrating your commits.

**Comment by apahim** [comment]
All merged.
